### PR TITLE
[IMP] stock_move_source_relocate: Code Refactor

### PR DESCRIPTION
**Pull request by mt-software-de**
Using float_compare instead of float_zero.
Also refactor tests for better use in dependency modules

### CODE REVIEW

**Comment by mt-software-de** [comment]
@jbaudoux and @rousseldenis could review this one?
It is needed for https://github.com/OCA/stock-logistics-warehouse/pull/1636

File: stock_move_source_relocate/models/stock_move.py [hunk 1]
```diff
@@ -35,37 +36,30 @@ def _apply_source_relocate(self):
             )
 
     def _apply_source_relocate_rule(self, relocation, reserved_availability, roundings):
-        relocated = self.env["stock.move"].browse()
-
+        self.ensure_one()
         rounding = roundings[self]
-        if not reserved_availability[self]:
+        qty_reserved = reserved_availability[self]
+        if float_compare(qty_reserved, 0, precision_rounding=rounding) <= 0:
```
Comment (rousseldenis) [inline stock_move_source_relocate/models/stock_move.py]: Could the qty_reserved really be < 0 ?

**Review by rousseldenis — APPROVED** [review]
Code review. Apart my question

File: stock_move_source_relocate/models/stock_move.py (hunk 1 above)
Comment (mt-software-de) [reply stock_move_source_relocate/models/stock_move.py]: I shouldn't be lower than 0. 
https://github.com/odoo/odoo/blob/7c105dc37bc0e8d740b86af3f9e88ddeb3c98f54/addons/stock/models/stock_move_line.py#L294

**Review by lmignon — APPROVED** [review]
LGTM (Code review only)

**Pull request merged**